File: optim_assign.py
import itertools
import random
import time

T1 = time.time()

A = [[0 for i in range(3)] for j in range(8)]

def score_sum(T):
        global A
        N = 0
        for i in range(8):
                N += A[i][int(T[i])]
        return N

print()
for i in range(8):
        B = A[i][0] + A[i][1]
        while B == 0 or B > 100 :
                A[i][0] = random.randrange(101)
                A[i][1] = random.randrange(101)
                B = A[i][0] + A[i][1]
        A[i][2] = 100 - B
        print("SCORE LIST " + str(i+1) + " : " + str(A[i]))
print()

C = list(map(lambda x : [score_sum(x), x], list(set(itertools.permutations("00011122")))))
C.sort(reverse=True) 

i = 0
D = int(C[0][0])
while True :
        print("MAX SCORE = " + str(D) + " : INDEX = " + str(C[i][1]))
        i += 1
        if D != C[i][0] :
                break

T2 = time.time()

print()
print("TIME : " + str(T2-T1))
print()

File: optim_assign.c

#include <algorithm>
#include <iostream>

#include <boost/random.hpp>
#include <boost/random/random_device.hpp>
#include <boost/timer/timer.hpp>

int score_array[21][3];

int main(){

	//boost::random::random_device seed_gen;
	boost::random::mt19937 gen;
	boost::random::uniform_int_distribution<> dist(0,100);

	using namespace std;

	boost::timer::cpu_timer timer;

	cout << "" << endl ;
	for(int i=0; i<21; i++){
		score_array[i][0] = 100 ;
		score_array[i][1] = 100 ;
		while(score_array[i][0] + score_array[i][1] > 100){
			score_array[i][0] = dist(gen);
			score_array[i][1] = dist(gen);
		}
		score_array[i][2] = 100 - (score_array[i][0] + score_array[i][1]) ;
		if(i < 9){
			cout << " SCORE LIST 0" << i+1 << " : [" << score_array[i][0] << ", " << score_array[i][1] << ", " << score_array[i][2] << "]" << endl ;
		}else{
			cout << " SCORE LIST "  << i+1 << " : [" << score_array[i][0] << ", " << score_array[i][1] << ", " << score_array[i][2] << "]" << endl ;
		}
	}
	cout << "" << endl ;
	
	int temp_sum ;
	string S = "000000011111112222222" ;
	int max_sum = 0 ;
	string max_index ;
	int C = 0 ;

	do{
		temp_sum = 0 ;
		for(int i=0; i<21; i++){
			temp_sum += score_array[i][int(S[i])-48];
		}
		if(temp_sum > max_sum){
			max_sum = temp_sum ;
			max_index = S ;
		}
		C++;
		if(C == 100000000){
			cout << S << endl ;
			C = 0 ;
		}
	}while(next_permutation(S.begin(), S.end()));

	cout << " MAX SCORE = " << max_sum << " : INDEX = " << max_index << endl ;
	cout << "" << endl ;

	std::string result = timer.format();
	cout << result << endl ;
	return 0 ;
}
